src/mscorlib/src/System/Threading/Overlapped.cs

```diff
@@ -94,7 +94,8 @@ static internal void IOCompletionCallback_Context(Object state)
                 overlapped = OverlappedData.GetOverlappedFromNative(pOVERLAP).m_overlapped;
                 helper = overlapped.iocbHelper;
 
-                if (helper == null || helper._executionContext == null || helper._executionContext == ExecutionContext.Default)
+                ExecutionContext context = helper?._executionContext;
+                if (context == null || context.IsDefault)
```

Comment:
If helper is null, don't we end up with an extra null check (checking context for null when we otherwise wouldn't)?  Or is the JIT able to remove it?

Reply:
No doesn't, will change